node/sync/src/block_sync.rs

```diff
@@ -936,12 +936,12 @@ impl<N: Network> BlockSync<N> {
     /// Removes block requests that have timed out, i.e, requests we sent that did not receive a response in time.
     ///
     /// This removes the corresponding block responses and returns the set of peers/addresses that timed out.
-    /// It will ask the communication service to ban any timed-out peers.
+    /// It will ask the peer pool handling service to ban any timed-out peers.
     ///
     /// Finally, it will return a set of new of block requests that replaced the timed-out requests (if needed).
-    pub fn handle_block_request_timeouts<C: CommunicationService>(
+    pub fn handle_block_request_timeouts<P: PeerPoolHandling<N>>(
         &self,
-        communication: &C,
+        peer_pool_handler: Option<&P>,
```

Comment:
this is a simple workaround to disable banning in tests without code duplication